src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs

```diff
@@ -47,21 +47,33 @@ public class RemoteUriWebService : IRemoteUriWebService {
                 }
             }
 
-            HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
-
-            if (context.Request.IsWebSocketRequest) {
-                Stream respStream = response.GetResponseStream();
-                string subProtocol = response.Headers[Constants.Headers.SecWebSocketProtocol];
-                var remoteWebSocket = CommonWebSocket.CreateClientWebSocket(respStream, subProtocol, TimeSpan.FromMinutes(10), 65335, true);
-                var websocketContext = await context.AcceptWebSocketAsync(subProtocol, 65335, TimeSpan.FromMinutes(10));
-                await WebSocketHelper.SendReceiveAsync(websocketContext.WebSocket, remoteWebSocket, ct);
-            } else {
-                SetResponseHeaders(response, context.Response, localBaseUrl, remoteBaseUrl);
-                using (Stream respStream = response.GetResponseStream())
-                using (Stream outStream = context.Response.OutputStream) {
-                    await outStream.CopyAndFlushAsync(respStream);
+            HttpWebResponse response = null;
+            try {
+                response = (HttpWebResponse)await request.GetResponseAsync();
+            } catch (WebException wex) {
+                if(wex.Status == WebExceptionStatus.ProtocolError) {
+                    response = wex.Response as HttpWebResponse;
+                } else {
+                    throw wex;
+                }
+            } finally {
+                if (response != null) {
+                    if (context.Request.IsWebSocketRequest) {
+                        Stream respStream = response.GetResponseStream();
+                        string subProtocol = response.Headers[Constants.Headers.SecWebSocketProtocol];
+                        var remoteWebSocket = CommonWebSocket.CreateClientWebSocket(respStream, subProtocol, TimeSpan.FromMinutes(10), 65335, true);
```

Comment:
Can you use named arguments here, so that it's clear what the number and the boolean correspond to?